Mirror/Runtime/NetworkProximityChecker.cs

```diff
@@ -93,10 +104,13 @@ public override bool OnRebuildObservers(HashSet<NetworkConnection> observers, bo
 
                     case CheckMethod.Physics2D:
                     {
-                        Collider2D[] hits = Physics2D.OverlapCircleAll(transform.position, visRange, castLayers);
-                        for (int i = 0; i < hits.Length; i++)
+                        // cast without allocating GC for maximum performance
+                        int hitCount = Physics2D.OverlapCircleNonAlloc(transform.position, visRange, hitsBuffer2D, castLayers);
+                        if (hitCount == hitsBuffer2D.Length) { Debug.LogWarning("NetworkProximityChecker's OverlapCircle test for " + name + " has filled the whole buffer(" + hitsBuffer2D.Length + "). Some results might have been omitted. Consider increasing buffer size."); }
```

Comment:
Use Array.Resize in that case

Reply:
we don't know what overlapcirclenonalloc will return, so we would have to resize and run it again afterwards.
possible, but might be a bit overkill here.
like you said in discord, I'll merge this for now and we can go from there.